Bug: syslog: test_basic_syslog_config failure due to mismatch in IPv6 case format

### Is it platform specific

generic

### Importance or Severity

Medium

### Description of the bug

Mgmt IPv6 gateway address configured in upper case in sonic-mgmt causes syslog testcase failure.

```
cat ansible/lab
  hosts:
    str-marvell-01:
      ansible_host: 10.28.60.9
      ansible_hostv6 : FC00:2::52
```

mgmt interface IP configuration on DUT:
```
    "MGMT_INTERFACE": {
        "eth0|10.28.60.9/24": {
            "forced_mgmt_routes": [
                "172.17.0.1/24"
            ],
            "gwaddr": "10.28.60.1"
        },
        "eth0|FC00:2::52/64": {
            "gwaddr": "fc00:2::1"
        }
    }
```

### Steps to Reproduce

Run syslog/test_syslog_source_ip.py tests
```
syslog/test_syslog_source_ip.py::TestSSIP::test_basic_syslog_config[str-marvell-01-None-vrf_set_source_set_None] 
```


### Actual Behavior and Expected Behavior

Actual behavior is test case failure:
```
syslog/test_syslog_source_ip.py::TestSSIP::test_basic_syslog_config[str-marvell-01-None-vrf_set_source_set_None] FAILED                                                             [100%]
```
Expected behavior is test case pass.

### Relevant log output

```shell
E               Failed: Excepted syslog config: server_ip fc00:2::1, source_ip FC00:2::52, vrf mgmt, port 800 doesn't exist

is_set_source = True
is_set_vrf = True
k          = 'ipv6'
port       = 800
self       = <tests.syslog.test_syslog_source_ip.TestSSIP object at 0x7f76cbd0f160>
source_ip  = 'FC00:2::52'
v          = {'source_ip': 'FC00:2::52/64', 'syslog_server_ip': 'fc00:2::1', 'syslog_server_mac': '90:90:90:90:90:32'}
vrf        = 'mgmt'
vrf_list   = ['default', 'Vrf-data', 'mgmt']

syslog/test_syslog_source_ip.py:375: Failed
```

### Output of `show version`

```shell
SONiC Software Version: SONiC.master.0-dirty-20250418.141456
SONiC OS Version: 12
Distribution: Debian 12.9
Kernel: 6.1.0-22-2-amd64
Build commit: fbf0ec145
Build date: Fri Apr 18 08:52:34 UTC 2025
Built by: rminnikanti@bglp-swdev086

Platform: x86_64-marvell_dbmvtx9180-r0
HwSKU: dbmvtx9180_64osfp_128x50G
ASIC: marvell-teralynx
ASIC Count: 1
Serial Number: N/A
Model Number: N/A
Hardware Revision: 0
Uptime: 07:49:45 up  3:34,  1 user,  load average: 1.51, 1.27, 1.05
Date: Thu 05 Dec 2024 07:49:45

Docker images:
REPOSITORY                       TAG                              IMAGE ID       SIZE
docker-dhcp-relay                latest                           5aa846f15ebd   321MB
docker-dhcp-relay                master.0-dirty-20250418.141456   5aa846f15ebd   321MB
docker-eventd                    latest                           57377ac3b90d   312MB
docker-eventd                    master.0-dirty-20250418.141456   57377ac3b90d   312MB
docker-sonic-bmp                 latest                           43888c5dd070   314MB
docker-sonic-bmp                 master.0-dirty-20250418.141456   43888c5dd070   314MB
docker-syncd-mrvl-teralynx-rpc   latest                           18e39d1a3577   654MB
docker-syncd-mrvl-teralynx-rpc   master.0-dirty-20250418.141456   18e39d1a3577   654MB
docker-macsec                    latest                           56c7443245b9   343MB
docker-macsec                    master.0-dirty-20250418.141456   56c7443245b9   343MB
docker-teamd                     latest                           2cc77decf4da   340MB
docker-teamd                     master.0-dirty-20250418.141456   2cc77decf4da   340MB
docker-sflow                     latest                           669eed11f8f6   341MB
docker-sflow                     master.0-dirty-20250418.141456   669eed11f8f6   341MB
docker-orchagent                 latest                           4f291d32a1c1   353MB
docker-orchagent                 master.0-dirty-20250418.141456   4f291d32a1c1   353MB
docker-nat                       latest                           bd42a37881e0   343MB
docker-nat                       master.0-dirty-20250418.141456   bd42a37881e0   343MB
docker-fpm-frr                   latest                           7c731d840edf   374MB
docker-fpm-frr                   master.0-dirty-20250418.141456   7c731d840edf   374MB
docker-sonic-mgmt-framework      latest                           75a1e45060f8   400MB
docker-sonic-mgmt-framework      master.0-dirty-20250418.141456   75a1e45060f8   400MB
docker-snmp                      latest                           09f290dba4f2   356MB
docker-snmp                      master.0-dirty-20250418.141456   09f290dba4f2   356MB
docker-router-advertiser         latest                           d632918e5e47   312MB
docker-router-advertiser         master.0-dirty-20250418.141456   d632918e5e47   312MB
docker-platform-monitor          latest                           cdcee5f55509   445MB
docker-platform-monitor          master.0-dirty-20250418.141456   cdcee5f55509   445MB
docker-mux                       latest                           3b1294223477   364MB
docker-mux                       master.0-dirty-20250418.141456   3b1294223477   364MB
docker-lldp                      latest                           772c114dc9b6   358MB
docker-lldp                      master.0-dirty-20250418.141456   772c114dc9b6   358MB
docker-sonic-gnmi                latest                           35e02b8840b6   425MB
docker-sonic-gnmi                master.0-dirty-20250418.141456   35e02b8840b6   425MB
docker-gnmi-watchdog             latest                           e285ac233747   312MB
docker-gnmi-watchdog             master.0-dirty-20250418.141456   e285ac233747   312MB
docker-framework                 latest                           e29e89227063   324MB
docker-framework                 master.0-dirty-20250418.141456   e29e89227063   324MB
docker-database                  latest                           31485b77af1a   319MB
docker-database                  master.0-dirty-20250418.141456   31485b77af1a   319MB
docker-auditd                    latest                           a3c894b0f864   312MB
docker-auditd                    master.0-dirty-20250418.141456   a3c894b0f864   312MB
docker-auditd-watchdog           latest                           b4bb07d46e53   312MB
docker-auditd-watchdog           master.0-dirty-20250418.141456   b4bb07d46e53   312MB
```

### Attach files (if any)

[master_syslog_failure_test.log](https://github.com/user-attachments/files/20231278/master_syslog_failure_test.log)

--- a/tests/syslog/test_syslog_source_ip.py
+++ b/tests/syslog/test_syslog_source_ip.py
@@ -375,8 +375,8 @@ class TestSSIP:
         """
         syslog_config_list = self.duthost.show_syslog()
         for syslog_config in syslog_config_list:
-            if all([syslog_config["server ip"] == syslog_server_ip,
-                    syslog_config["source ip"] == source,
+            if all([syslog_config["server ip"].lower() == syslog_server_ip.lower(),
+                    syslog_config["source ip"].lower() == source.lower(),
                     syslog_config["vrf"] == vrf,
                     syslog_config["port"] == port]):
                 return True

